[CalloutCard] Update title prop to accept ReactNode type (#7684)
Shopify/polaris: .changeset/funny-moose-hug.md

---
'@shopify/polaris': minor
---

> Modify CalloutCard title prop to accept ReactNode type

Reviewer: ```suggestion
Update the `CalloutCard` `title` prop to accept a ReactNode type
```
